.github/workflows/main.yml

```diff
@@ -57,82 +73,106 @@ jobs:
           os: ubuntu-latest
           rust: stable
           other: i686-unknown-linux-gnu
+          if: always()
         - name: Linux x86_64 beta
           os: ubuntu-latest
           rust: beta
           other: i686-unknown-linux-gnu
+          if: ${{ needs.channel.outputs.CHANNEL == 'beta' || needs.channel.outputs.CHANNEL == 'nightly' }}
         - name: Linux x86_64 nightly
           os: ubuntu-latest
           rust: nightly
           other: i686-unknown-linux-gnu
+          if: ${{ needs.channel.outputs.CHANNEL == 'nightly' }}
         - name: macOS x86_64 stable
           os: macos-latest
           rust: stable
           other: x86_64-apple-ios
+          if: always()
         - name: macOS x86_64 nightly
           os: macos-latest
           rust: nightly
           other: x86_64-apple-ios
+          if: ${{ needs.channel.outputs.CHANNEL == 'nightly' }}
         - name: Windows x86_64 MSVC stable
           os: windows-latest
           rust: stable-msvc
           other: i686-pc-windows-msvc
+          if: always()
         - name: Windows x86_64 gnu nightly # runs out of space while trying to link the test suite
           os: windows-latest
           rust: nightly-gnu
           other: i686-pc-windows-gnu
+          if: ${{ needs.channel.outputs.CHANNEL == 'nightly' }}
     name: Tests ${{ matrix.name }}
     steps:
     - uses: actions/checkout@v3
+      if: matrix.if
```

Comment:
Unfortunately `job.if` cannot access fields in `matrix`. We cannot simply write `if: matrix.if` for `job` and instead we need to do it in each `step`. That's terrible. Totally understand if someone objects to this PR due to that.

See https://github.com/actions/runner/issues/1985

Reply:
Ugh, that is really unfortunate.  Is it possible to use something like https://stackoverflow.com/a/65434401 to have a job that runs first to generate the matrix? I'm not sure if that's any better, since it is also a pretty complex solution.

Reply:
Fixed with 4ff82fdc391a2b06ae3c9bc4630e0314428a5a18. Doesn't look too bad though.